Orso.Arpa.Application/Localization/LocalizerCache.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Orso.Arpa.Domain.Entities;
+using Orso.Arpa.Domain.Interfaces;
+
+namespace Orso.Arpa.Application.Localization
+{
+    public class LocalizerCache
+    {
+        private static IServiceCollection _services;
+        private static readonly object _syncLock = new();
+        private static List<Translation> _translations;
+
+        public LocalizerCache(IServiceCollection services)
+        {
+            _services = services;
+        }
+
+        public Task CallBack()
+        {
+            return Task.Run(() => _translations = GetDbTranslationList());
+        }
+
+        public string GetTranslation(string key, string resourceKey, string culture)
+        {
+
+            if (_translations.IsNullOrEmpty())
+            {
+                _translations = GetDbTranslationList();
+                return key;
+            }
+
+            Translation translation = _translations.AsQueryable().DefaultIfEmpty(null).FirstOrDefault(d =>
+                d.Deleted == false && d.Key == key && d.ResourceKey == resourceKey && Regex.Match(d.LocalizationCulture, "(^|[^-])"+culture+"([^-]|$)", RegexOptions.None).Success);
+
+            return (translation == null) ? key : translation.Text ?? key;
+        }
+
+        public virtual IList<Translation> GetAllTranslations(string culture, string resourceKey)
+        {
+            if (_translations.IsNullOrEmpty())
+                return new List<Translation>();
+
+            IQueryable<Translation> query = _translations.AsQueryable().Where(d =>
+                d.Deleted == false && d.ResourceKey.Equals(resourceKey) && Regex.Match(d.LocalizationCulture, "(^|[^-])"+culture+"([^-]|$)", RegexOptions.None).Success);
```

Comment:
Auch hier - Muss hier wirklich Deleted nochmal geprüft werden? Das Regex Pattern hast du hier schon zum zweiten Mal verwendet. Vielleicht ausgliedern?

Reply:
Regex habe ich als Konstante eingefügt.
Deleted kann ich hier entfernen, muss es jedoch weiter unten in die Query einfügen, da ich den QueryFilter entfernt habe.